cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -197,10 +197,13 @@ public synchronized void delete(final Id.Namespace namespaceId) throws Exception
       // Another reason for not deleting the default namespace is that we do not want to call a delete on the default
       // namespace in the storage provider (Hive, HBase, etc), since we re-use their default namespace.
       if (!Id.Namespace.DEFAULT.equals(namespaceId)) {
-        // Finally delete namespace from MDS
-        nsStore.delete(namespaceId);
-        // Delete namespace in storage providers
-        dsFramework.deleteNamespace(namespaceId);
+        try {
+          // Delete namespace in storage providers
+          dsFramework.deleteNamespace(namespaceId);
+        } finally {
+          // Finally delete namespace from MDS
+          nsStore.delete(namespaceId);
```

Comment:
@anwar6953 Yes HDFS happens first. But preferenceStore.deleteProperties and other HBase related operations are performed before this. So if HBase is down, those wouldn't have succeeded either.